wandb/apis/public.py

```diff
@@ -2449,12 +2449,16 @@ def more(self):
     @property
     def cursor(self):
         if self.last_response:
-            return self.last_response["project"]["run"][self.run_key]["edges"]["cursor"]
+            return self.last_response["project"]["run"][self.run_key]["edges"][-1][
+                "cursor"
+            ]
         else:
             return None
 
     def update_variables(self):
-        self.variables.update({"cursor": self.cursor})
+        self.variables.update(
+            {"cursor": self.cursor, "artifactLimit": self.per_page,}
+        )
```

Comment:
This method isnt really adding value, if you use "perPage" instead of artifactLimit, as done in other derived classes then you can just use the baseclass implementation of update_variables.  (See ArtifactVersions)